op-node/rollup/derive/span_batch.go

```diff
@@ -505,15 +509,20 @@ func (b *SpanBatch) peek(n int) *SpanBatchElement { return b.Batches[len(b.Batch
 // updates l1OriginCheck or parentCheck if needed.
 func (b *SpanBatch) AppendSingularBatch(singularBatch *SingularBatch, seqNum uint64) error {
 	// if this new element is not ordered with respect to the last element, panic
-	if len(b.Batches) > 0 && b.peek(0).Timestamp > singularBatch.Timestamp {
+	if len(b.Batches) > 0 && b.peek(0).Timestamp >= singularBatch.Timestamp {
 		panic("span batch is not ordered")
 	}
 
 	// always append the new batch and set the L1 origin check
 	b.Batches = append(b.Batches, singularBatchToElement(singularBatch))
 
 	// always update the L1 origin check
+	copy(b.lastL1OriginCheck[:], b.L1OriginCheck[:])
 	copy(b.L1OriginCheck[:], singularBatch.EpochHash.Bytes()[:20])
+
+	// it's easy to tend to only update lastParentCheck when ParentCheck is updated,
+	// but it's wrong because if done that way, lastParentCheck will always be empty.
+	copy(b.lastParentCheck[:], b.ParentCheck[:])
```

Comment:
Because `ParentCheck` will only be updated once, so if `lastParentCheck` is also updated only once, it'll be the initial state of  `ParentCheck`, which is empty. I don't have a good sentence that comes to my mind, so I deleted this comment.